src/blockaxis.jl

```diff
@@ -0,0 +1,184 @@
+
+# interface
+getindex(b::AbstractVector, K::BlockIndex{1}) = b[Block(K.I[1])][K.α[1]]
+getindex(b::AbstractVector, K::BlockIndexRange{1}) = b[K.block][K.indices[1]]
+
+function findblockindex(b::AbstractVector, k::Integer)
+    K = findblock(b, k)
+    K[searchsortedfirst(b[K], k)] # guaranteed to be in range
+end
+
+function _BlockedUnitRange end
+
+
+"""
+    BlockedUnitRange
+
+is an `AbstractUnitRange{Int}` that has been divided
+into blocks, and is used to represent axes of block arrays.
+Construction is typically via `blockrange` which converts
+a vector of block lengths to a `BlockedUnitRange`.
+```jldoctest; setup = quote using BlockArrays end
+julia> blockedrange([2,2,3])
+3-blocked 7-element BlockedUnitRange{Array{Int64,1}}:
+ 1
+ 2
+ ─
+ 3
+ 4
+ ─
+ 5
+ 6
+ 7
+```
+"""    
+struct BlockedUnitRange{CS} <: AbstractUnitRange{Int}
+    first::Int
+    cumsum::CS
```

Comment:
I wonder if using `lasts` instead of `cumsum` makes sense.  Isn't it a bit more descriptive than `cumsum`?

Reply:
Changed!